src/auto_contiguous.cpp

```diff
@@ -12,6 +12,13 @@ void auto_contiguous::apply(module& p) const
 {
     for(auto ins : iterator_for(p))
     {
+        // apply to sub_modules
+        const auto& mod_inputs = ins->module_inputs();
+        for(const auto& mod : mod_inputs)
+        {
+            this->apply(*mod);
+        }
```

Comment:
Passes should not call out to the submodules directly. Rather run_passes should call the passes for the child modules before calling it for the parent module.(I believe this is the reverse of what `get_sub_modules` return).

Reply:
Changed